--- configure.in~	Mon Nov  8 15:06:12 1999
+++ configure.in	Wed Jan 19 14:37:50 2000
@@ -1903,10 +1903,10 @@
 
 dnl MesaGL mesagl
 
-dnl test -z "$mesagl_direct_test_library" && mesagl_direct_test_library=MesaGLU
+dnl test -z "$mesagl_direct_test_library" && mesagl_direct_test_library=GLU
 AC_DEFUN(AC_PATH_MESAGL_DIRECT,
-[test -z "$mesagl_direct_test_library" && mesagl_direct_test_library=MesaGL
-test -z "$mesagl_direct_test_library" && mesagl_direct_test_library=MesaGLU
+[test -z "$mesagl_direct_test_library" && mesagl_direct_test_library=GL
+test -z "$mesagl_direct_test_library" && mesagl_direct_test_library=GLU
 test -z "$mesagl_direct_test_function" && mesagl_direct_test_function=glXCreateContext
 test -z "$mesagl_direct_test_include" && mesagl_direct_test_include=GL/gl.h
   for ac_dir in               \
@@ -2057,7 +2057,7 @@
   if test "$gl" = no; then
     gl=yes
     AC_DEFINE(USE_GL)
-    XLOCKLIBS="${XLOCKLIBS} -lMesaGL -lMesaGLU"
+    XLOCKLIBS="${XLOCKLIBS} -lGL -lGLU"
     test "x$mesagl_includes" = xNONE && mesagl_includes=$ac_mesagl_includes
     test "x$mesagl_libraries" = xNONE && mesagl_libraries=$ac_mesagl_libraries
     case "${canonical}" in
@@ -2073,8 +2073,8 @@
     case "${canonical}" in
       *-*-linux* )
         # Testing if MesaGL was compiled against VGA
-        if test -f $mesagl_libraries/libMesaGL.so; then
-          have_mesagl_svga=`nm $mesagl_libraries/libMesaGL.so | grep SVGAMesa | wc -l`
+        if test -f $mesagl_libraries/libGL.so; then
+          have_mesagl_svga=`nm $mesagl_libraries/libGL.so | grep SVGAMesa | wc -l`
         else
           have_mesagl_svga=0
         fi
