diff -Nru xlockmore-5.00.orig/xlock/XLock.ad xlockmore-5.00/xlock/XLock.ad
--- xlockmore-5.00.orig/xlock/XLock.ad	Mon Dec  4 15:29:37 2000
+++ xlockmore-5.00/xlock/XLock.ad	Wed May 15 13:03:15 2002
@@ -56,9 +56,9 @@
 !XLock.program: fortune -o
 XLock.program: fortune -s
 !XLock.program: fortune
-!XLock.program: /usr/games/fortune -o
-!XLock.program: /usr/games/fortune -s
-!XLock.program: /usr/games/fortune
+!XLock.program: /usr/bin/fortune -o
+!XLock.program: /usr/bin/fortune -s
+!XLock.program: /usr/bin/fortune
 !XLock.program: finger
 !XLock.messagesfile: fortune.dat
 !XLock.messagefile: /etc/hosts
