diff -Naur xlockmore-4.17.2/configure.in xlockmore-4.17.2-p/configure.in
--- xlockmore-4.17.2/configure.in	Tue Oct 17 11:08:16 2000
+++ xlockmore-4.17.2-p/configure.in	Tue Oct 17 11:08:32 2000
@@ -2116,7 +2116,7 @@
   AC_DEFINE(USE_DPMS)
 dnl test needed for Digital Unix machines
   if test "x$ac_dpms_libraries" != xNONE; then
-    XLOCKLIBS="${XLOCKLIBS} -lXdpms"
+    XLOCKLIBS="${XLOCKLIBS}"
   fi
   test "x$dpms_includes" = xNONE && dpms_includes=$ac_dpms_includes
   test "x$dpms_libraries" = xNONE && dpms_libraries=$ac_dpms_libraries
